package.xml

```diff
@@ -9,6 +9,8 @@
 
   <author email="jacob@openrobotics.org">Jacob Perron</author>
 
+  <exec_depend>ros_environment</exec_depend>
```

Comment:
In general, I actually think this is a good thing; it will make the error message much more useful.

However, looking through the core, the only other package that currently depends on `ros_environment` is `ros2doctor`.  I'm fine with adding this as a dependency, but pinging @ros2/team to get thoughts on any possible problems from doing this.

Reply:
AFAIK, the page we're linking to isn't different across distros (though I guess that could change). That being said, I can't think of any problem that might arise from depending on ros_environment.

Reply:
> AFAIK, the page we're linking to isn't different across distros (though I guess that could change).

Yeah, true.  The thing that is nice about this is that it links directly into the distribution you are using, so the logo at the top is correct and all of the links are into the correct version of the document.

Reply:
Actually, from a conceptual perspective, I think this isn't desirable, if `rpyutils` is anything like `rcutils` or `rcpputils`. Both of those intentionally do not depend on or provide any "ROS" related things. They are just meant to provide generic utilities that are _used_ by ROS.

That being said, in practice it's probably fine but it also means that it's just slightly less likely people might use it outside of ROS, but that's a hypothetical benefit.

Reply:
Good point! At least in this case, we're not explicitly importing any Python module, rather checking the env variable typically set by ros_environment.

Reply:
> Good point! At least in this case, we're not explicitly importing any Python module, rather checking the env variable typically set by ros_environment.

That's true.  From that perspective, we could leave out the `exec_depend` on `ros_environment`, and just hope that the environment variable is set.  If it isn't, we have the fallback to 'rolling'.

That said, that is also conceptually tying this implementation to ROS as well, so I don't know.

Reply:
To be clear, I'm fine with adding this dependency, but I just wanted to point out the conceptual issue.

Reply:
the first `r` in `rpyutils`/`rcutils`/`rcpputils` refers to "ROS" so it's all tied to ROS anyway :stuck_out_tongue_winking_eye:

> we could leave out the exec_depend on ros_environment, and just hope that the environment variable is set. If it isn't, we have the fallback to 'rolling'.

That's a good alternative/compromise. If we had a `/latest` redirect (i.e. -> `/galactic` currently) it would be even better I think, but we don't.

Reply:
Yeah, we've debated back and forth about having that `/latest` link.  One practical problem is that we can't actually use `/latest`; that is currently being used by ROS 1.  We could use `/latest_ros2` or something, but it lacks a certain ring.

Anyway, I don't think we should hold this up on that particular point.  I'm fine with just dropping the `exec_depend`, but I'd like to hear from @wjwwood and @jacobperron before doing that.

Reply:
> the first r in rpyutils/rcutils/rcpputils refers to "ROS" so it's all tied to ROS anyway 😜

Yes, but they are things _used_  by ROS, in order to implement ROS. The point was always that they never _depend_ on ROS, firstly to avoid circular dependencies and second so they could be reused outside the project.

For example, people widely use `apr` as a utility library, even though it is the _Apache_ Portable Runtime. I see these as the same, i.e. they come out of the ROS project, but do not depend on it.

Reply:
Again, that's all philosophical, in practice the dependency on `ros_environment` does not introduce a circular dependency and so I don't mind if we have it there.

Reply:
> We could use `/latest_ros2` or something, but it lacks a certain ring.

This is a bit off topic now, but what about `/latest2`? I just thought I'd mention it. It looks like someone just hit the `2` key on their keyboard by accident, but it's simple.

Reply:
I lean towards remove the exec_depend, just because it's not strictly required. But I'm okay with this PR either way.

Reply:
> This is a bit off topic now, but what about `/latest2`? I just thought I'd mention it. It looks like someone just hit the `2` key on their keyboard by accident, but it's simple.

That's true, but there is some question about what the semantics about `latest2` means.  Is it the latest LTS (that would be Foxy right now)?  The latest released distribution (that would be Galactic right now)?  The latest that keeps moving forward (that would be Rolling)?

Anyway, I'll let this part of the discussion stop here.  @christophebedard if you'd like to continue talking about this link, then I'll suggest opening an issue against https://github.com/ros2/ros2_documentation, and we can continue the discussion there.